kork/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/artifactstore/ArtifactStore.java

```diff
@@ -28,20 +32,44 @@ public class ArtifactStore implements ArtifactStoreGetter, ArtifactStoreStorer {
 
   private final ArtifactStoreStorer artifactStoreStorer;
 
+  private final Map<String, List<ApplicationStorageFilter>> exclude;
+
   public ArtifactStore(
-      ArtifactStoreGetter artifactStoreGetter, ArtifactStoreStorer artifactStoreStorer) {
+      ArtifactStoreGetter artifactStoreGetter,
+      ArtifactStoreStorer artifactStoreStorer,
+      Map<String, List<ApplicationStorageFilter>> exclude) {
     this.artifactStoreGetter = artifactStoreGetter;
     this.artifactStoreStorer = artifactStoreStorer;
+    this.exclude = exclude;
+  }
+
+  public boolean shouldExclude(String type, String application) {
+    return application == null
+        || this.exclude.containsKey(type)
+            && this.exclude.get(type).stream().anyMatch((filter) -> filter.filter(application));
   }
 
   /** Store an artifact in the artifact store */
-  public Artifact store(Artifact artifact) {
-    return artifactStoreStorer.store(artifact);
+  public Artifact store(Artifact artifact, ArtifactDecorator... decorators) {
```

Comment:
hrmm, shouldn't storage independent of app be allowed?  CERTAIN operations do operate independent of an application (mostly a few orchestration tasks).  NOT saying no :) just more wondering about this particular logic.  OR if the app should be "default" or some other thing in those cases.  NOT a "no" just me thinking about if there's certain operations here.

Reply:
Application is a requirement for artifact storage.

The reference urls are `ref://<application>/<hash>`. The reason we did this is to ensure that applications are not sharing artifacts which could get dangerous very quickly.

Reply:
yeah mostly wondering if there should be a "default" similar to the way that RBAC uses an "anonymous" hidden user when no defaults are set.  This would allow "non-application" tasks to operate kinda thing. NOT that there are many of those...

Reply:
There wasnt a use case for it at least for us, but it would trivial to do if anyone needs it.